### DISCUSSION TITLE
open admin console page and got infinite loading

### DISCUSSION THREAD

**Discussion by choguun** [deeztek/Hermes-Secure-Email-Gateway#46] [Support]
Hi Deeztek team,

I get problem when open admin console page that infinite loading.

This site can’t be reached
172.31.2.47 took too long to respond.
Try:

Checking the connection
[Checking the proxy and the firewall](chrome-error://chromewebdata/#buttons)
ERR_CONNECTION_TIMED_OUT

you can try this

https://3.0.202.102/admin/

**Comment by deeztek** [comment]
It looks like it's redirecting to https://172.31.2.47/admin/ which is presumably the private IP you have assigned to the system which is the expected behavior. What happens when you try to directly load **https://172.31.2.47/admin/**

**Reply by choguun** [reply]
It cannot load directly to 172.31.2.47 because it was an instance that I remote via ssh. How do I config to use public IP?
Thank you for getting back to me so quickly.

**Reply by deeztek** [reply]
Sure no problem. So, am I to understand that the system has two IPs assigned to it? In other words, how to you SSH into the box? Using the public IP or the private IP? Just trying to figure out how the Hermes install script picked up the private IP address.

**Reply by choguun** [reply]
I use AWS like this "ubuntu@ec2-3-0-202-102.ap-southeast-1.compute.amazonaws.com"

**Reply by deeztek** [reply]
ah ok. So you access via the elastic IP on AWS. Ok this is going to be a bit involved but not too bad. 

**WARNING: It goes without saying but I'll say it anyway, I take no responsibility for whatever happens to your system, your cat, your dog,  your house etc.. by following the instructions below.**

So first and foremost you need to decide on a public FQDN that you will access Hermes SEG from. You could use the "[ec2-3-0-202-102.ap-southeast-1.compute.amazonaws.com](mailto:ubuntu@ec2-3-0-202-102.ap-southeast-1.compute.amazonaws.com)" FQDN, but I would advise against that because you will probably have issues with 3rd party certificates but that's up to you. If you do decide to use a different FQDN that's not the one AWS assigned to you, ensure that public DNS for that FQDN points to the elastic IP of your instance. The important thing to remember is that you will **only** be able to use a FQDN to access the Hermes Web Console from this point on.

**Important: Before you make the changes below, ensure the public FQDN points to the elastic IP of your instance**

1. Edit the following files and replace every instance of your private IP with the public FQDN you decided upon and then save them:

- /etc/nginx/snippets/auth.conf
- /etc/authelia/configuration.yml

2. Restart the following services:

`systemctl restart nginx`
`systemctl restart authelia`

3. You should now be able to access Hermes SEG Web Console using the public FQDN you set above and login with the default credentials (admin/ChangeMe2!):

[https://[PUBLIC_FQDN]/admin/](https://[PUBLIC_FQDN]/admin/)

4. Next, navigate to **System --> Network Settings** and set the network mode to DHCP or Static. The only caveat with DHCP is the fact that AWS specifies a MAC address in the network config and Hermes will not configure the **macaddress:** parameter. So, I would probably go with Static mode and set the IP to the private IP of your instance, set the subnet mask, set the gateway, set DNS etc.. according to your environment. Ensure you set the Hostname and the Primary Domain Name to the public FQDN you set above, save and apply. This will set the hostname/domain name system wide as necessary but will again break the Hermes Web Console.

5. AGAIN, edit the following files and replace every instance of your private IP with the public FQDN you decided upon and then save them:

- /etc/nginx/snippets/auth.conf
- /etc/authelia/configuration.yml

6. Restart the following services:

`systemctl restart nginx`
`systemctl restart authelia`

You will have to re-enter the [https://[PUBLIC_FQDN]/admin/](https://[PUBLIC_FQDN]/admin/) FQDN in your browser because the system will again try to redirect to the private IP address as before.

7. Next navigate to **System --> Console Settings** and change the Console Mode to Host Name. Ensure you set the Host Name to the public FQDN you set above and Submit. 

8. Follow the [Getting Started Guide](https://docs.deeztek.com/books/hermes-seg-administrator-guide/page/getting-started) to get everything setup correctly. Most importantly, **ensure you change the passwords for the admin system account and the Ciphermail admin account**. We also **HIGHLY** recommend you set 2FA for the system account.

**Reply by deeztek** [reply]
@choguun Please see additional edits to the post above

**Reply by choguun** [reply]
Thanks for the prompt reply. I follow number 1,2,3 then I log in /admin I get this message 

"We apologize for the inconvenience but a system error has occurred or malicious activity has been detected or database credentials are missing/incorrect

Page Specific Error below
Appplication.cfc: session is unauthorized

Attempting to modify a URL parameter or form input or attempting to use certain keywords such as (select, update etc.) in form fields will generate this error. Please go back to retry your operation without modifying any system URLs and/or change your form input with fields that are not banned. Additionally, verify the database credentials are correct under System --> System Settings.

If you have any further questions, please contact support"

Not sure I do it correctly?

Thank you for support me

**Reply by deeztek** [reply]
I suspect that's due to the fact that the system console mode does not jive with Authelia. Okay, let's try this. Download this script in the SSH console and run it. Ensure you pick FQDN mode and type the public FQDN when prompted:

[https://gitlab.deeztek.com/dedwards/hermes-seg-18.04/-/raw/b1a70fe9078e470bd5ef80965284ed8bc37d4140/dirstructure/opt/hermes/scripts/set_console_mode_manual.sh?inline=false](https://gitlab.deeztek.com/dedwards/hermes-seg-18.04/-/raw/b1a70fe9078e470bd5ef80965284ed8bc37d4140/dirstructure/opt/hermes/scripts/set_console_mode_manual.sh?inline=false)

**Reply by deeztek** [reply]
One more thing, set the following in your /etc/hosts file:

127.0.0.1       PUBLIC_FQDN

where PUBLIC_FQDN is the public FQDN you set above.

BTW, I hate AWS

**Reply by choguun** [reply]
Following your suggestion got the same problem 
"We apologize for the inconvenience but a system error has occurred or malicious activity has been detected or database credentials are missing/incorrect...". 

Below is the query result from the parameters2 table from my server.

MariaDB [hermes]> select * from parameters2;
+-----+-----------------------------------------------+----------------------+--------------+--------+---------+
| id  | parameter                                     | value2               | module       | active | applied |
+-----+-----------------------------------------------+----------------------+--------------+--------+---------+
|   1 | server_ip_octet1                              | 192                  | network      |      1 |       1 |
|   2 | server_ip_octet2                              | 168                  | network      |      1 |       1 |
|   3 | server_ip_octet3                              | 69                   | network      |      1 |       1 |
|   4 | server_ip_octet4                              | 254                  | network      |      1 |       1 |
|   5 | server_subnet                                 | 255.255.255.0        | network      |      1 |       1 |
|  25 | server_dns1_octet2                            | 168                  | network      |      1 |       1 |
|  26 | server_dns1_octet3                            | 69                   | network      |      1 |       1 |
|   9 | server_network_octet1                         | NULL                 | network      |      1 |       1 |
|  10 | server_network_octet2                         | NULL                 | network      |      1 |       1 |
|  11 | server_network_octet3                         | NULL                 | network      |      1 |       1 |
|  12 | server_network_octet4                         | NULL                 | network      |      1 |       1 |
|  13 | server_broadcast_octet1                       | NULL                 | network      |      1 |       1 |
|  14 | server_broadcast_octet2                       | NULL                 | network      |      1 |       1 |
|  15 | server_broadcast_octet3                       | NULL                 | network      |      1 |       1 |
|  16 | server_broadcast_octet4                       | NULL                 | network      |      1 |       1 |
|  17 | server_gateway_octet1                         | 192                  | network      |      1 |       1 |
|  18 | server_gateway_octet2                         | 168                  | network      |      1 |       1 |
|  19 | server_gateway_octet3                         | 69                   | network      |      1 |       1 |
|  20 | server_gateway_octet4                         | 1                    | network      |      1 |       1 |
|  21 | server_dns1_octet1                            | 192                  | network      |      1 |       1 |
|  22 | network_mode                                  | dhcp                 | network      |      1 |       1 |
|  23 | server_name                                   | hermes               | network      |      1 |       1 |
|  24 | server_domain                                 | domain.tld           | network      |      1 |       1 |
|  27 | server_dns1_octet4                            | 1                    | network      |      1 |       1 |
|  28 | server_dns2_octet1                            |                      | network      |      1 |       1 |
|  29 | server_dns2_octet2                            |                      | network      |      1 |       1 |
|  30 | server_dns2_octet3                            |                      | network      |      1 |       1 |
|  31 | server_dns2_octet4                            |                      | network      |      1 |       1 |
|  32 | server_dns3_octet1                            |                      | network      |      1 |       1 |
|  33 | server_dns3_octet2                            |                      | network      |      1 |       1 |
|  34 | server_dns3_octet3                            |                      | network      |      1 |       1 |
|  35 | server_dns3_octet4                            |                      | network      |      1 |       1 |
|  36 | firewall_status                               | disabled             | firewall     |      1 |       1 |
|  55 | system_log_retention                          | 30                   | systemlog    |      1 |       1 |
|  37 | ScanMail                                      | true                 | clamav       |      1 |       1 |
|  38 | ScanArchive                                   | true                 | clamav       |      1 |       1 |
|  39 | ArchiveBlockEncrypted                         | false                | clamav       |      1 |       1 |
|  40 | ScanPE                                        | true                 | clamav       |      1 |       1 |
|  41 | ScanOLE2                                      | true                 | clamav       |      1 |       1 |
|  42 | ScanHTML                                      | true                 | clamav       |      1 |       1 |
|  43 | AlgorithmicDetection                          | true                 | clamav       |      1 |       1 |
|  44 | ScanELF                                       | true                 | clamav       |      1 |       1 |
|  45 | PhishingSignatures                            | true                 | clamav       |      1 |       1 |
|  46 | PhishingScanURLs                              | true                 | clamav       |      1 |       1 |
|  47 | PhishingAlwaysBlockSSLMismatch                | false                | clamav       |      1 |       1 |
|  48 | PhishingAlwaysBlockCloak                      | false                | clamav       |      1 |       1 |
|  49 | DetectPUA                                     | false                | clamav       |      1 |       1 |
|  50 | HeuristicScanPrecedence                       | false                | clamav       |      1 |       1 |
|  51 | OLE2BlockMacros                               | false                | clamav       |      1 |       1 |
|  52 | ScanPDF                                       | true                 | clamav       |      1 |       1 |
|  59 | default_action                                | accept               | dkim         |      1 |       1 |
|  60 | dkim_enable                                   | no                   | dkim         |      1 |       1 |
|  61 | body_canonicalization                         | relaxed              | dkim         |      1 |       1 |
|  62 | badsignature_action                           | accept               | dkim         |      1 |       1 |
|  63 | dnserror_action                               | tempfail             | dkim         |      1 |       1 |
|  64 | internalerror_action                          | tempfail             | dkim         |      1 |       1 |
|  65 | nosignature_action                            | accept               | dkim         |      1 |       1 |
|  66 | security_action                               | tempfail             | dkim         |      1 |       1 |
|  67 | signature_algorithm                           | rsa-sha256           | dkim         |      1 |       1 |
|  68 | headers_canonicalization                      | relaxed              | dkim         |      1 |       1 |
|  69 | server_ip                                     | 192.168.69.254       | network      |      1 |       1 |
|  70 | server_gateway                                | 192.168.69.1         | network      |      1 |       1 |
|  71 | server_dns1                                   | 192.168.69.1         | network      |      1 |       1 |
|  72 | server_dns2                                   |                      | network      |      1 |       1 |
|  73 | server_dns3                                   |                      | network      |      1 |       1 |
|  74 | server_broadcast                              |                      | network      |      1 |       1 |
|  75 | debugLevel                                    | 1                    | spf          |      1 |       1 |
|  76 | TestOnly                                      | 1                    | spf          |      1 |       1 |
|  77 | HELO_reject                                   | Fail                 | spf          |      1 |       1 |
|  78 | Mail_From_reject                              | Fail                 | spf          |      1 |       1 |
|  79 | PermError_reject                              | False                | spf          |      1 |       1 |
|  80 | TempError_Defer                               | False                | spf          |      1 |       1 |
|  81 | FailureReports                                | false                | dmarc        |      1 |       1 |
|  82 | RejectFailures                                | false                | dmarc        |      1 |       1 |
|  83 | report_email                                  | NULL                 | dmarc        |      1 |       1 |
|  84 | report_org                                    | NULL                 | dmarc        |      1 |       1 |
|  85 | interval                                      | daily                | dmarc        |      1 |       1 |
|  86 | startdate                                     | NULL                 | dmarc        |      1 |       1 |
|  87 | starttime                                     | NULL                 | dmarc        |      1 |       1 |
|  88 | jwt_secret                                    | q1zfymLDClc6e1ZTv60w | authelia     |      1 |       1 |
|  89 | access_control.rules.policy                   |                      | authelia     |      1 |       1 |
|  90 | authentication_backend.disable_reset_password | false                | authelia     |      1 |       1 |
|  91 | session.name                                  | hermes_session       | authelia     |      1 |       1 |
|  92 | session.secret                                | OT6CZydvPIwtODtQb74G | authelia     |      1 |       1 |
|  93 | session.expiration                            | 3600                 | authelia     |      1 |       1 |
|  94 | session.inactivity                            | 3600                 | authelia     |      1 |       1 |
|  95 | session.domain                                | NULL                 | authelia     |      1 |       1 |
|  96 | notifier.smtp.host                            | [127.0.0.1]          | authelia     |      1 |       1 |
|  97 | notifier.smtp.port                            | 10026                | authelia     |      1 |       1 |
|  98 | notifier.smtp.sender                          | no-reply@domain.tld  | authelia     |      1 |       1 |
|  99 | notifier.smtp.subject                         | [Hermes SEG] {title} | authelia     |      1 |       1 |
| 100 | regulation.max_retries                        | 5                    | authelia     |      1 |       1 |
| 101 | regulation.find_time                          | 120                  | authelia     |      1 |       1 |
| 102 | regulation.ban_time                           | 300                  | authelia     |      1 |       1 |
| 103 | log.level                                     | debug                | authelia     |      1 |       1 |
| 104 | log.format                                    | text                 | authelia     |      1 |       1 |
| 105 | access_control.domain                         | domain.tld           | authelia     |      1 |       1 |
| 106 | console.certificate                           | 1                    | console      |      1 |       2 |
| 107 | smtp.certificate                              | 1                    | certificates |      1 |       1 |
| 109 | console.host                                  | 3.0.202.102          | console      |      1 |       2 |
| 108 | console.mode                                  | ip                   | console      |      1 |       2 |
| 133 | console.dhparam                               | disable              | console      |      1 |       2 |
| 134 | console.hsts                                  | disable              | console      |      1 |       2 |
| 135 | console.ssl_stapling                          | disable              | console      |      1 |       2 |
| 136 | console.ssl_stapling_verify                   | disable              | console      |      1 |       2 |
| 137 | HoldQuarantinedMessages                       | false                | dmarc        |      1 |       1 |
+-----+-----------------------------------------------+----------------------+--------------+--------+---------+

**Reply by deeztek** [reply]
Thank you for query results. Very helpful. As you can see in the results the following fields are wrong:

```
| 109 | console.host | 3.0.202.102 | console | 1 | 2 |
| 108 | console.mode | ip | console | 1 | 2 |
```

The console.mode needs to be **fqdn** and the console.host needs to be the public FQDN. Were you able to download this script in the SSH console and run it, ensuring you picked FQDN mode and type the public FQDN when prompted?

[https://gitlab.deeztek.com/dedwards/hermes-seg-18.04/-/raw/b1a70fe9078e470bd5ef80965284ed8bc37d4140/dirstructure/opt/hermes/scripts/set_console_mode_manual.sh?inline=false](https://gitlab.deeztek.com/dedwards/hermes-seg-18.04/-/raw/b1a70fe9078e470bd5ef80965284ed8bc37d4140/dirstructure/opt/hermes/scripts/set_console_mode_manual.sh?inline=false)

Did you also add the following in your /etc/hosts file?

`127.0.0.1 PUBLIC_FQDN`

where PUBLIC_FQDN is the public FQDN you set above.

**Reply by deeztek** [reply]
If you still having issues, we can help with a remote support session. If interested click the link below and pick a date/time that works for you for a zoom session:

https://calendso.deeztek.com/support@deeztek.com/60-minute-support-session

Thanks
